File: solutions/go/word-count/1/word_count.go
package wordcount

import "strings"

type Frequency map[string]int

func WordCount(phrase string) Frequency {
	phrase = strings.ToLower(phrase)
	phrase = strings.Replace(phrase, ",", " ", -1)
	phrase = strings.Replace(phrase, ".", " ", -1)
	phrase = strings.Replace(phrase, ":", " ", -1)
	words := strings.Fields(phrase)

	for i := 0; i < len(words); i++ {
		words[i] = strings.Trim(words[i], ".,!$%^&*()_+-=<>?/|:;\"'@")
	}

	freq := make(Frequency)
	for _, word := range words {
		freq[word]++
	}
	return freq
}

File: solutions/java/word-count/1/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

class WordCount {
    public Map<String, Integer> phrase(String input) {

        Map<String, Integer> result = new HashMap<>();
        input = input.toLowerCase().replaceAll("[^a-z1-9']|\\B'|'\\B", " ").trim();
        var words = input.split("\\s+");

        for (String word : words) {
            var wordCount = result.get(word);
            result.put(word, wordCount == null ? 1 : wordCount + 1);
        }

        return result;
    }
}
